kernel/src/scan/mod.rs

```diff
@@ -490,11 +655,12 @@ fn transform_to_logical_internal(
                         "logical schema did not contain expected field, can't transform data",
                     ));
                 };
-                let name = field.physical_name(global_state.column_mapping_mode)?;
+                let name = field.physical_name();
```

Comment:
@nicklan -- any idea why we didn't just apply column mapping to the partition columns along with all the others? Why wait until now?